pkg/services/quota/quota.go

```diff
@@ -2,201 +2,12 @@ package quota
-				return true, err
-			}
-			if query.Result.Limit < 0 {
-				continue
-			}
-			if query.Result.Limit == 0 {
-				return true, nil
-			}
-
-			if query.Result.Used >= query.Result.Limit {
-				return true, nil
-			}
-		case "user":
-			if scopeParams == nil || scopeParams.UserId == 0 {
-				continue
-			}
-			query := models.GetUserQuotaByTargetQuery{UserId: scopeParams.UserId, Target: scope.Target, Default: scope.DefaultLimit, UnifiedAlertingEnabled: qs.Cfg.UnifiedAlerting.IsEnabled()}
-			if err := qs.SQLStore.GetUserQuotaByTarget(ctx, &query); err != nil {
-				return true, err
-			}
-			if query.Result.Limit < 0 {
-				continue
-			}
-			if query.Result.Limit == 0 {
-				return true, nil
-			}
-
-			if query.Result.Used >= query.Result.Limit {
-				return true, nil
-			}
-		}
-	}
-	return false, nil
-}
-
-func (qs *QuotaService) getQuotaScopes(target string) ([]models.QuotaScope, error) {
-	scopes := make([]models.QuotaScope, 0)
-	switch target {
-	case "user":
-		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.User},
-			models.QuotaScope{Name: "org", Target: "org_user", DefaultLimit: qs.Cfg.Quota.Org.User},
-		)
-		return scopes, nil
-	case "org":
-		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.Org},
-			models.QuotaScope{Name: "user", Target: "org_user", DefaultLimit: qs.Cfg.Quota.User.Org},
-		)
-		return scopes, nil
-	case "dashboard":
-		scopes = append(scopes,
-			models.QuotaScope{
-				Name:         "global",
-				Target:       target,
-				DefaultLimit: qs.Cfg.Quota.Global.Dashboard,
-			},
-			models.QuotaScope{
-				Name:         "org",
-				Target:       target,
-				DefaultLimit: qs.Cfg.Quota.Org.Dashboard,
-			},
-		)
-		return scopes, nil
-	case "data_source":
-		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.DataSource},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.DataSource},
-		)
-		return scopes, nil
-	case "api_key":
-		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.ApiKey},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.ApiKey},
-		)
-		return scopes, nil
-	case "session":
-		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.Session},
-		)
-		return scopes, nil
-	case "alert_rule": // target need to match the respective database name
-		scopes = append(scopes,
-			models.QuotaScope{Name: "global", Target: target, DefaultLimit: qs.Cfg.Quota.Global.AlertRule},
-			models.QuotaScope{Name: "org", Target: target, DefaultLimit: qs.Cfg.Quota.Org.AlertRule},
-		)
-		return scopes, nil
-	default:
-		return scopes, ErrInvalidQuotaTarget
-	}
+	DeleteUser(context.Context, int64) error
```

Comment:
it's not clear what it's supposed to do; I would suggest renaming it to something like `DeleteUserQuota`

Reply:
i tried to follow the naming convention we decided on in the services, so when called it would be something like `quotaService.DeleteUser()`

Reply:
I'm fine with the `DeleteByUser`